machine_learning/benchmarks/circuit_qnn_classifier_benchmark.py

```diff
@@ -73,7 +70,59 @@ def parity(x):
         self.circuit_classifier_fitted = NeuralNetworkClassifier(
             neural_network=self.circuit_qnn, optimizer=COBYLA()
         )
-        self.circuit_classifier_fitted.fit(self.X, self.y01)
+        self.circuit_classifier_fitted.fit(X, y)
+
+    def setup_dataset_iris(self, X, y, quantum_instance_name):
+        """Training CircuitQNN function for iris dataset."""
+
+        self.output_shape = 3
+
+        # scaling data
+        scaler = MinMaxScaler((-1, 1))
+        self.X = scaler.fit_transform(X)
+
+        # creating feature map
+        feature_dim = self.X.shape[1]
+        feature_map = ZFeatureMap(feature_dim)
+
+        # creating ansatz
+        ansatz = RealAmplitudes(feature_dim)
+
+        qc = QuantumCircuit(feature_dim)
+        qc.append(feature_map, range(feature_dim))
+        qc.append(ansatz, range(feature_dim))
+
+        def three_class(x):
+            return f"{x:b}".count("1") % 3
+
+        # construct QNN
+        circuit_qnn = CircuitQNN(
+            circuit=qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            interpret=three_class,
+            output_shape=self.output_shape,
+            quantum_instance=self.backends[quantum_instance_name],
+        )
+
+        self.circuit_classifier_fitted = NeuralNetworkClassifier(
+            neural_network=circuit_qnn,
+            optimizer=NELDER_MEAD(maxiter=10),
```

Comment:
Same here, why `NELDER_MEAD` here and `COBYLA` in the synthetic case?
In general, setup should be identical in the similar tests, easier to compare.

Reply:
Updated so both synthetic and iris datasets use Nelder-Mead